tpm/constants.go

```diff
@@ -71,6 +76,63 @@ const (
 	tpmCapFlagPermanent    uint32 = 0x00000108
 )
 
+// permission type
+type permission uint32
+
+// NV Permissions and Operations
+// Note: Permissions are summable
+const (
+	nvPerPPWrite    permission = 0x00000001
+	nvPerOwnerWrite permission = 0x00000002
+	nvPerAuthWrite  permission = 0x00000004
+	nvPerWriteAll   permission = 0x00000800
+	// Warning: The Value 0x00001000 is
+	// defined in the spec as
+	// TPM_NV_PER_WRITEDEFINE, but it is
+	// not included directly in this
+	// code because it locks the given
+	// NV Index permanently if used
+	// incorrectly. This operation can't
+	// be undone in any way. Do not use
+	// this value unless you know what
+	// you're doing!
+	nvPerWriteSTClear permission = 0x00002000
+	nvPerGlobalLock   permission = 0x00004000
+	nvPerPPRead       permission = 0x00008000
+	nvPerOwnerRead    permission = 0x00100000
+	nvPerAuthRead     permission = 0x00200000
+	nvPerReadSTClear  permission = 0x80000000
+)
+
+// permMap : Map of TPM_NV_Permissions to its strings for convenience
+var permMap = map[permission]string{
+	nvPerPPWrite:      "PPWrite",
+	nvPerOwnerWrite:   "OwnerWrite",
+	nvPerAuthWrite:    "AuthWrite",
+	nvPerWriteAll:     "WriteAll",
+	nvPerWriteSTClear: " WriteSTClear",
+	nvPerGlobalLock:   "GlobalLock",
+	nvPerPPRead:       "PPRead",
+	nvPerOwnerRead:    "OwnerRead",
+	nvPerAuthRead:     "AuthRead",
+	nvPerReadSTClear:  "ReadSTClear",
+}
+
+// String returns a textual representation of the set of permissions
+func (p permission) String() string {
+	var retString strings.Builder
+	for iterator, item := range permMap {
+		if (p & iterator) != 0 {
+			retString.WriteString(item + " + ")
+		}
+	}
+	if retString.String() == "" {
+		return "Permission/s not found"
+	}
+	return retString.String()
+
+}
```

Comment:
Looks good!

One nit: it might not be ideal to always add the ` + ` to the end, because if you only have two permissions, you end up with this: `OwnerRead + AuthRead + `.

The easiest way to fix this would be:

```go
func (p permission) String() string {
	var retString strings.Builder
	for iterator, item := range permMap {
		if (p & iterator) != 0 {
			retString.WriteString(item + " + ")
		}
	}
	if retString.String() == "" {
		return "Permission/s not found"
	}
	return strings.TrimSuffix(retString.String(), " + ")
}
```

If you prefer, another pattern to build this string is by keeping a slice of the different pieces & joining them at the end using `strings.Join()`:

```go
// String returns a textual representation of the set of permissions
func (p permission) String() string {
	var out []string
	for iterator, item := range permMap {
		if (p & iterator) != 0 {
			out = append(out, item)
		}
	}
	if len(out) == 0 {
		return "Permission/s not found"
	}
	return strings.Join(out, " + ")
}
```